app/src/main/java/com/sample/appconnectsample/LoginActivity.java

```diff
@@ -53,6 +45,28 @@ public void doLogInButton(View source) {
         new LogInTask().execute(username, password);
     }
 
+    public void doRegisterButton(View source) {
+        Intent intent = new Intent(LoginActivity.this, RegistrationEmailActivity.class);
+        startActivityForResult(intent, RegistrationEmailActivity.REGISTRATION_REQUEST, null);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (resultCode == RESULT_OK && requestCode == RegistrationEmailActivity.REGISTRATION_REQUEST) {
+            // The user registered successfully, set the email field, and clear password
+            usernameField.setText(data.getStringExtra("email"));
+            passwordField.setText("");
+
+            passwordField.requestFocus();
+
+            new AlertDialog.Builder(LoginActivity.this).
```

Comment:
Added a popup to inform that registration is successful, and password field is highlighted for entry, email field pre populated.